### PR TITLE
golang.org/x/text bump

### PR DESCRIPTION
**Pull request by Bslabe123**
# Description

Bumped x/text to v0.3.8 to fix https://github.com/advisories/GHSA-69ch-w2m2-3vjp

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: changelog/v1.9.28/x-text-bump.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+changelog:
+  - type: DEPENDENCY_BUMP
+    dependencyOwner: golang.org
+    dependencyRepo: x/text
+    dependencyTag: v0.3.8
+    issueLink: https://github.com/solo-io/gloo/issues/7364
+    resolvesIssue: true
```
Comment (nfuden) [inline changelog/v1.9.28/x-text-bump.yaml:7]: Definitely does not resolve

File: go.mod [hunk 2]
```diff
@@ -132,8 +132,7 @@ replace (
 	github.com/renstrom/dedent => github.com/lithammer/dedent v1.0.0
 	golang.org/x/net v0.0.0-20211112202133-69e39bad7dc2 => golang.org/x/net v0.0.0-20211209124913-491a49abca63
 	golang.org/x/net v0.0.0-20211205041911-012df41ee64c => golang.org/x/net v0.0.0-20211209124913-491a49abca63
-
-	golang/x/text => golang/x/text v0.3.7
```
[the remark this answers is not in the document]
Comment (Bslabe123) [reply go.mod:137]: It would require a go-utils bump, do we want to do this?

File: changelog/v1.9.28/x-text-bump.yaml (hunk 1 above)
Comment (Bslabe123) [reply changelog/v1.9.28/x-text-bump.yaml:7]: But theres only one CVE in the issue and its fixed? Sorry I may be missing something here

Comment (nfuden) [reply changelog/v1.9.28/x-text-bump.yaml:7]: As long as we don't forget the ee ticket and merge

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (nfuden) [reply go.mod:137]: Ah so the only source is a dep we don't want to mess with? I guess that's fine if the go utils doesn't have a quick update in patch versions

**Pull request closed without merging**